ui/src/time_picker.tsx

```diff
@@ -0,0 +1,259 @@
+import DateFnsUtils from '@date-io/date-fns'
+import { VirtualElement } from '@popperjs/core/lib'
+import { CalendarOrClockPickerView } from '@mui/x-date-pickers/internals/models'
+
+/**
+ * Create a time picker.
+ *
+ * A time picker allows a user to pick a time value.
+ */
+
+export interface TimePicker {
+  /** An identifying name for this component. */
+  name: Id
+  /** Text to be displayed alongside the component. */
+  label?: S
+  /** A string that provides a brief hint to the user as to what kind of information is expected in the field. */
+  placeholder?: S
+  /** The time value in hh:mm format. E.g. '10:30', '14:25', '23:59', '00:00' */
+  value?: S
+  /** True if this field is disabled. */
+  disabled?: B
+  /** The width of the time picker, e.g. '100px'. Defaults to '100%'. */
+  width?: S
+  /** True if the component should be visible. Defaults to True. */
+  visible?: B
+  /** True if the form should be submitted when the time is selected. */
+  trigger?: B
+  /** True if this is a required field. Defaults to False. */
+  required?: B
+  /** Specifies 12-hour or 24-hour time format. One of `12` or `24`. Defaults to `24`. */
+  hour_format?: S
+  /** The minimum allowed time value in hh:mm format. E.g.: '08:00', '13:30' */
+  min?: S
+  /** The maximum allowed time value in hh:mm format. E.g.: '15:30', '00:00' */
+  max?: S
+  /** Limits the available minutes to select from. One of `1`, `5`, `10`, `15`, `20`, `30` or `60`. Defaults to `1`. */
+  minutes_step?: U
+}
+
+const
+  css = stylesheet({
+    toolbar: {
+      paddingTop: 16,
+      paddingLeft: 16
+    },
+    toolbarTime: {
+      fontSize: 26,
+      cursor: 'pointer',
+      $nest: {
+        '&:hover': {
+          backgroundColor: cssVar('$neutralLighter'),
+          color: cssVar('$neutralDark')
+        }
+      }
+    },
+    toolbarText: { fontSize: 26 },
+    toolbarLabel: { maxWidth: '70%' }
+  }),
+  popoverProps: Partial<PopperProps> | undefined = {
+    placement: 'bottom-start',
+    sx: {
+      '& .MuiPaper-root': {
+        borderRadius: '2px',
+        boxShadow: `${cssVar('$text1')} 0px 6.4px 14.4px 0px, ${cssVar('$text2')} 0px 1.2px 3.6px 0px`
+      },
+      '& .MuiTypography-root, [role=listbox]>span': {
+        fontFamily: '"Inter", -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif',
+      }
+    }
+  }
+
+type ToolbarProps = { time: S, setOpenView: (view: CalendarOrClockPickerView) => void, label: S | undefined, switchAmPm: () => void }
+
+const
+  ThemeProvider = React.lazy(() => import('@mui/material/styles').then(({ ThemeProvider }) => ({ default: ThemeProvider }))),
+  LocalizationProvider = React.lazy(() => import('@mui/x-date-pickers').then(({ LocalizationProvider }) => ({ default: LocalizationProvider }))),
+  TimePicker = React.lazy(() => import('@mui/x-date-pickers').then(({ TimePicker }) => ({ default: TimePicker }))),
```

Comment:
It would be better if we could import just the stuff we need instead of whole modules. Would something like this work?

```suggestion
  ThemeProvider = React.lazy(() => import('@mui/material/styles/ThemeProvider').then(({ ThemeProvider }) => ({ default: ThemeProvider }))),
  LocalizationProvider = React.lazy(() => import('@mui/x-date-pickers/LocalizationProvider').then(({ LocalizationProvider }) => ({ default: LocalizationProvider }))),
  TimePicker = React.lazy(() => import('@mui/x-date-pickers/TimePicker').then(({ TimePicker }) => ({ default: TimePicker }))),
```

Reply:
It works, but not with jest. It looks like jest cannot read these files.

```
console.error node_modules/jest-environment-jsdom-fourteen/node_modules/jsdom/lib/jsdom/virtual-console.js:29
    Error: Uncaught [SyntaxError: Unexpected token 'export']
        at reportException (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/jest-environment-jsdom-fourteen/node_modules/jsdom/lib/jsdom/living/helpers/runtime-script-errors.js:62:24)
        at innerInvokeEventListeners (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/jest-environment-jsdom-fourteen/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:331:9)
        at invokeEventListeners (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/jest-environment-jsdom-fourteen/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:266:3)
        at HTMLUnknownElementImpl._dispatch (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/jest-environment-jsdom-fourteen/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:212:11)
        at HTMLUnknownElementImpl.dispatchEvent (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/jest-environment-jsdom-fourteen/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:87:17)
        at HTMLUnknownElement.dispatchEvent (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/jest-environment-jsdom-fourteen/node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js:144:23)
        at Object.invokeGuardedCallbackDev (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/react-dom/cjs/react-dom.development.js:3994:16)
        at invokeGuardedCallback (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/react-dom/cjs/react-dom.development.js:4056:31)
        at beginWork$1 (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/react-dom/cjs/react-dom.development.js:23964:7)
        at performUnitOfWork (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/react-dom/cjs/react-dom.development.js:22779:12) Jest encountered an unexpected token
    
    This usually means that you are trying to import a file which Jest cannot parse, e.g. it's not plain JavaScript.
    
    By default, if Jest sees a Babel config, it will use that to transform your files, ignoring "node_modules".
    
    Here's what you can do:
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.
    
    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/en/configuration.html
    
    Details:
    
    /Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/@mui/material/styles/ThemeProvider.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){export { ThemeProvider as default } from '@mui/system';
                                                                                             ^^^^^^
    
    SyntaxError: Unexpected token 'export'
        at new Script (vm.js:102:7)
        at ScriptTransformer._transformAndBuildScript (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/@jest/transform/build/ScriptTransformer.js:537:17)
        at ScriptTransformer.transform (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/@jest/transform/build/ScriptTransformer.js:579:25)
        at Runtime._execModule (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/jest-runtime/build/index.js:842:53)
        at Runtime._loadModule (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/jest-runtime/build/index.js:577:12)
        at Runtime.requireModule (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/jest-runtime/build/index.js:433:10)
        at Runtime.requireModuleOrMock (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/node_modules/jest-runtime/build/index.js:598:21)
        at /Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/src/time_picker.tsx:96:36
        at processTicksAndRejections (internal/process/task_queues.js:95:5)

  console.error node_modules/react-dom/cjs/react-dom.development.js:20085
    The above error occurred in one of your React components:
    
        at Lazy
        at Suspense
        at XTimePicker (/Users/mmihok/Documents/h2o-ai/projects/wave/repo/wave/ui/src/time_picker.tsx:132:27)
    
    Consider adding an error boundary to your tree to customize error handling behavior.
    Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
```

Reply:
The problem is not in reading, but in parsing the files - need to be transformed first via `babel` (https://stackoverflow.com/questions/60714101/how-to-setup-jest-with-node-modules-that-use-es6). However, seems like this is only a problem for ThemeProvider, the rest of lazy components work fine.

Please add TODO to the ThemeProvider then so it's not forgotten and we can move ahead with the release today.

Reply:
UPDATE: The problem causes just `import('@mui/material/styles/ThemeProvider')` and even the bundle is little bit larger. Other two imports `import('@mui/x-date-pickers/LocalizationProvider')` and  `import('@mui/x-date-pickers/TimePicker')` work perfect and are reducing bundle size (`40+ kB`).

Reply:
Sorry, your message above was not displayed while writing last message. I'll add TODO.